azafea/event_processors/endless/metrics/events/__init__.py

```diff
@@ -555,6 +555,112 @@ def _get_fields_from_payload(payload: GLib.Variant) -> Dict[str, Any]:
         }
 
 
+class ParentalControlsBlockedFlatpakInstall(SingularEvent):
+    __tablename__ = 'parental_controls_blocked_flatpak_install'
+    __event_uuid__ = '9d03daad-f1ed-41a8-bc5a-6b532c075832'
+    __payload_type__ = 's'
+
+    app = Column(Unicode, nullable=False)
+
+    @staticmethod
+    def _get_fields_from_payload(payload: GLib.Variant) -> Dict[str, Any]:
+        return {
+            'app': payload.get_string(),
+        }
+
+
+class ParentalControlsBlockedFlatpakRun(SingularEvent):
+    __tablename__ = 'parental_controls_blocked_flatpak_run'
+    __event_uuid__ = 'afca2515-e9ce-43aa-b355-7663c770b4b6'
+    __payload_type__ = 's'
+
+    app = Column(Unicode, nullable=False)
+
+    @staticmethod
+    def _get_fields_from_payload(payload: GLib.Variant) -> Dict[str, Any]:
+        return {
+            'app': payload.get_string(),
+        }
+
+
+class ParentalControlsChanged(SingularEvent):
+    __tablename__ = 'parental_controls_changed'
+    __event_uuid__ = '449ec188-cb7b-45d3-a0ed-291d943b9aa6'
+    __payload_type__ = 'a{sv}'
+
+    # The a{sv} contains the fields from
+    # `com.endlessm.ParentalControls.AppFilter`. The a{sv} describes the new
+    # parental controls settings for one user. It may also contain the following
+    # additional optional fields:
+    #  - IsAdministrator (`b`): whether the user is an administrator
+    #  - IsInitialSetup (`b`): whether this event is happening during gnome-initial-setup
+    app_filter_is_whitelist = Column(Boolean, nullable=False)
+    app_filter = Column(ARRAY(Unicode, dimensions=1), nullable=False)
```

Comment:
One could imagine instead maintaining a table of app IDs and having an intermediate table to join that to this event. But let's do this simpler thing for now.

Reply:
:+1: